app/controllers/labels_controller.rb

```diff
@@ -10,24 +10,32 @@ class LabelsController < ApplicationController
   def index; end
 
   def update
-    @label.update(params.permit(:color))
+    if @label.update(params.permit(:color))
+      render json: @label.to_h
+    else
+      flash.now[:alert] = @label.errors.full_messages
+      render json: @label.errors.full_messages, status: :unprocessable_entity
+    end
   end
 
   def destroy
     @label.destroy
   end
 
-  def merge
-    label_ids = params[:label_ids]
+  def merge # rubocop:disable Metrics/AbcSize
+    label_ids = params[:label_ids] || []
 
     ApplicationRecord.transaction do
-      Label.find(label_ids.first).update!(
-        name: params[:new_label_name],
-        tasks: Task.where(id: TaskLabel.select(:task_id).where(label_id: label_ids))
-      )
+      tasks = Task.where(id: TaskLabel.select(:task_id).where(label_id: label_ids))
       Label.destroy(label_ids[1..])
-    rescue StandardError => e
-      render json: e, status: :unprocessable_entity
+      Label.find(label_ids.first).update!(name: params[:new_label_name], tasks:)
+    rescue ActiveRecord::RecordNotFound => e
+      flash.now[:alert] = e.message
+      render json: e.message, status: :unprocessable_entity
```

Comment:
If the record is not found, we might also return a 404, or specifically a `:not_found` -- or would you stay with the 422 (which would be fine as well)?

Reply:
I think it is okay to leave it as is. When one or more of the labels are not found the merge query is not processable as a whole.

Reply:
Alright, fine for me. Thanks!